test/e2e/data/infrastructure-docker/main/clusterclass-quick-start.yaml

```diff
@@ -440,21 +483,38 @@ spec:
       - containerPath: "/var/run/docker.sock"
         hostPath: "/var/run/docker.sock"
 ---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: DockerMachinePoolTemplate
+metadata:
+  name: quick-start-default-worker-machinepooltemplate
+  labels:
+    InfraMachineTemplate.machinePool.label: "InfraMachineTemplate.machinePool.labelValue"
+  annotations:
+    InfraMachineTemplate.machinePool.annotation: "InfraMachineTemplate.machinePool.annotationValue"
+spec:
+  template:
+    spec:
+      template: {}
+---
 apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
 kind: KubeadmConfigTemplate
 metadata:
   name: quick-start-default-worker-bootstraptemplate
   labels:
     BootstrapConfigTemplate.machineDeployment.label: "BootstrapConfigTemplate.machineDeployment.labelValue"
+    BootstrapConfigTemplate.machinePool.label: "BootstrapConfigTemplate.machinePool.labelValue"
```

Comment:
As mentioned above let's have separate KubeadmConfigTemplates for md/mp and only set the labels/annotations on the right ones